pennylane/fourier/spectrum.py

```diff
@@ -228,7 +245,10 @@ def wrapper(*args, **kwargs):
                 # inputs can only enter one-parameter gates
                 continue
 
-            inpt = op.parameters[0]
+            # extract the parameter as a numpy scalar value,
+            # since some tensor objects are unhashable and
+            # can therefore not be used as dictionary keys
+            inpt = qml.math.toarray(op.parameters[0])
```

Comment:
This is still not satisfying, since the user will need to make sure now that the value is unique...Since we never really compute anything with the params this is fine, but it is a big "gotcha"